src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/ImportController.php

```diff
@@ -0,0 +1,217 @@
+            'import_file_names' => $names,
+            'import_dir' => $this->get('prestashop.import.dir')->getDir(),
+            'max_file_upload_size' => $this->getPostMaxSizeInBytes(),
+        ];
+    }
+
+    /**
+     * Handle import file upload
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function uploadAction(Request $request)
+    {
+        $uploadedFile = $request->files->get('file');
+        if (!$uploadedFile instanceof UploadedFile) {
+            return $this->json([
+                'error' => $this->trans('No file was uploaded.', 'Admin.Advparameters.Notification')
+            ]);
+        }
+
+        try {
+            $fileUploader = $this->get('prestashop.import.file_uploader');
+            $file = $fileUploader->upload($uploadedFile);
+        } catch (FileUploadException $e) {
+            $response['error'] = $e->getMessage();
+
+            return $this->json($response);
+        }
+
+        $response['file'] = [
+            'name' => $file->getFilename(),
+            'size' => $file->getSize(),
+        ];
+
+        return $this->json($response);
+    }
+
+    /**
+     * Delete import file
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction(Request $request)
+    {
+        if ($filename = $request->query->get('filename')) {
+            $fileRemoval = $this->get('prstashop.import.file_removal');
+            $fileRemoval->remove($filename);
+        }
+
+        return $this->redirectToRoute('admin_import');
+    }
+
+    /**
+     * Download import file from history
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function downloadAction(Request $request)
+    {
+        if ($filename = $request->query->get('filename')) {
+            $importDirectory = $this->get('prestashop.import.dir');
+
+            $response = new BinaryFileResponse($importDirectory.$filename);
+            $response->setContentDisposition(
+                ResponseHeaderBag::DISPOSITION_ATTACHMENT,
+                $filename
+            );
+
+            return $response;
+        }
+
+        return $this->redirectToRoute('admin_import');
+    }
+
+    /**
+     * Fowards submitted form data to legacy import page
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    private function fowardRequestToLegacyResponse(Request $request)
+    {
+        $legacyController = $request->attributes->get('_legacy_controller');
+        $legacyContext =  $this->get('prestashop.adapter.legacy.context');
+
+        $legacyImportUrl = $legacyContext->getAdminLink($legacyController);
+
+        return $this->redirect($legacyImportUrl, Response::HTTP_TEMPORARY_REDIRECT);
+    }
+
+    /**
+     * Get m post max size from ini configuration in bytes
+     *
+     * @return int
+     */
+    private function getPostMaxSizeInBytes()
```

Comment:
would you mind to extract it as a service? I have the feeling we may re-use it at some point (and for other properties)